index.md

```diff
@@ -8,6 +8,469 @@ This is some welcome text.
+docstring. Students are expected to write complete docstrings for every module, 
+function, class and method.
+
+```python
+def is_true(obj):
+    """
+    """
+    if obj is False:
+        return False
+    else:
+        return True
+```
+
+### C0113 Unneeded not {#C0113}
+
+### C0121 Singleton comparison {#C0121}
+
+This is an error that occurs when an expression is compared to singleton values 
+like True, False or None. This type of error indicates incorrect comparison 
+tests, so please check your code carefully.
+
+```python
+def is_true():
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    temp = 5
+    if temp == True:
+        return False
+    else:
+        return True
+```
+
+### C0122 Misplaced comparison constant {#C0122}
+
+This error occurs when a constant is placed on the left side of a comparison. 
+It is usually a better practice to place it in the right side of the comparison.
+
+```python
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if 0 > obj:
+        return False
+    else:
+        return True
+```
+
+### C0123 Unidiomatic type check {#C0123}
+
+This error occurs when type() is used instead of isinstance() for a type check. Students are expected to use isinstance(x, Y) instead of type(x) == Y. The example below demonstrates the same.
+
+```python
+def is_int(obj):
+    """Check is the given object is of type 'int'.
+    
+    @type obj: object
+    @rtype: bool
+    """
+    return type(obj) == "<type 'int'>"
+```
+
+The above can be modified to:
+
+```python
+def is_int(obj):
+    """Check is the given object is of type 'int'.
+    
+    @type obj: object
+    @rtype: bool
+    """
+    return isinstance(obj, int)
+```
+
+### E0101 Return in init {#E0101}
+
+This error occurs when a return statement is used in the __init__ method. 
+Students are expected to not return anything in the __init__ method.
+
+```python
+class Animal:
+    """An animal in the zoo.
+    """
+    
+    def __init__(self, name):
+        self.name = name
+        return True
+        
+    def is_animal(self):
+        """
+        @rtype: bool
+        """
+        return isinstance(self.name, str)
+        
+    def getter(self):
+        """
+        @rtype: str
+        """
+        return self.name
+```
+
+### E0102 Function redefined {#E0102}
+
+This error occurs when a function, class or method is redefined. If you are 
+getting this error, you should make sure all the functions, methods and classes 
+have different names.
+
+```python
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if obj < 0:
+        return False
+    else:
+        return True
+        
+        
+def pos(obj):
+    """
+    @type obj: int
+    @rtype: bool
+    """
+    if obj < 1:
+        return False
+    else:
+        return True
+```
+
+### E0103 Not in loop {#E0103}
+
+This error occurs when break and continue keywords are used outside loops. If 
```

Comment:
As above, the second sentence is redundant and should be omitted.

Explain briefly why these keywords belong inside the loop.